### ISSUE TITLE
Player only plays for one second.

### ISSUE THREAD

**Issue by doriansgithub** [youtube/youtube-ios-player-helper#443]
Player only plays for one second. If I pause the process, it plays normal. Any ideas?

UserInfo={NSLocalizedFailureReason=(originator doesn't have entitlement com.apple.runningboard.assertions.webkit AND originator doesn't have entitlement com.apple.multitasking.systemappassertions)}

**Comment by steven-worrall** [comment]
I'm having a very similar problem, the player will load up fine but once I try to play the content I get a bunch of errors. 

[assertion] Error acquiring assertion: <Error Domain=RBSServiceErrorDomain Code=1 "(originator doesn't have entitlement com.apple.runningboard.assertions.webkit AND originator doesn't have entitlement com.apple.multitasking.systemappassertions)" UserInfo={NSLocalizedFailureReason=(originator doesn't have entitlement com.apple.runningboard.assertions.webkit AND originator doesn't have entitlement com.apple.multitasking.systemappassertions)}>

[ProcessSuspension] 0x10a3b30f0 - ProcessAssertion: Failed to acquire RBS assertion 'WebKit Media Playback' for process with PID=74905, error: Error Domain=RBSServiceErrorDomain Code=1 "(originator doesn't have entitlement com.apple.runningboard.assertions.webkit AND originator doesn't have entitlement com.apple.multitasking.systemappassertions)" UserInfo={NSLocalizedFailureReason=(originator doesn't have entitlement com.apple.runningboard.assertions.webkit AND originator doesn't have entitlement com.apple.multitasking.systemappassertions)}

**Comment by doriansgithub** [comment]
Exactly the same here.

**Comment by gondp** [comment]
Same problem. Solution? iOS 15.1

**Comment by developingbrian** [comment]
> Same problem. Solution? iOS 15.1

Im seeing the same errors on 15.1 unfortunately.

**Comment by schradeyannik** [comment]
Same here

**Comment by numerized** [comment]
same problem here

**Comment by dj-fiorex** [comment]
Same problem for me, with this library and with another project that use plain web audio api.

**Comment by themcaguy** [comment]
Did anyone find any solution to this problem? I am also facing the same.

**Comment by saltyskip** [comment]
Any updates?

**Comment by CanfieldMobileDev** [comment]
My app is having these problems playing videos inside a web view running on a webpage supplied by a 3rd party.

While researching this issue and not finding any answers, only questions, I have run into this [article](https://eclecticlight.co/2019/11/07/runningboard-a-new-subsystem-in-catalina-to-detect-errors). It directly addresses the apparent culprit - "RunningBoard: a new subsystem in Catalina to detect errors".

I am going over it now. Perhaps others also can try to make sense of it.

**Comment by Skylark4Sky** [comment]
same here

**Comment by villyGreen** [comment]
Same problems

**Comment by daniellivingston** [comment]
Solution: set `NSAllowsArbitraryLoadsInWebContent` to `true`.

In your `Info.plist` file, enable the property list key [`NSAllowsArbitraryLoadsInWebContent`](https://developer.apple.com/documentation/bundleresources/information_property_list/nsapptransportsecurity/nsallowsarbitraryloadsinwebcontent) as follows:

```xml
	<key>NSAllowsArbitraryLoadsInWebContent</key>
	<true/>
```

If that doesn't work, try going overboard and disabling all relevant App Transport Security restrictions (and then, of course, re-enabling all non-breaking security measures once the problem is isolated):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSAllowsArbitraryLoadsForMedia</key>
	<true/>
	<key>NSAllowsArbitraryLoadsInWebContent</key>
	<true/>
	<key>NSAllowsArbitraryLoads</key>
	<true/>
	<key>NSAllowsLocalNetworking</key>
	<true/>
</dict>
</plist>
```

**Comment by shto** [comment]
> Solution: set `NSAllowsArbitraryLoadsInWebContent` to `true`.
> 
> In your `Info.plist` file, enable the property list key [`NSAllowsArbitraryLoadsInWebContent`](https://developer.apple.com/documentation/bundleresources/information_property_list/nsapptransportsecurity/nsallowsarbitraryloadsinwebcontent) as follows:
> 
> ```
> 	<key>NSAllowsArbitraryLoadsInWebContent</key>
> 	<true/>
> ```
> 
> If that doesn't work, try going overboard and disabling all relevant App Transport Security restrictions (and then, of course, re-enabling all non-breaking security measures once the problem is isolated):
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
> 	<key>NSAllowsArbitraryLoadsForMedia</key>
> 	<true/>
> 	<key>NSAllowsArbitraryLoadsInWebContent</key>
> 	<true/>
> 	<key>NSAllowsArbitraryLoads</key>
> 	<true/>
> 	<key>NSAllowsLocalNetworking</key>
> 	<true/>
> </dict>
> </plist>
> ```

@daniellivingston Enabled all, but still getting the assertion.

**Comment by daniellivingston** [comment]
@shto 
Sorry, I misspoke. They should be in your `Entitlements.plist`, not `Info.plist`. If you don't have an Entitlements file, you can add & manipulate one under *Target -> Capabilities*: https://stackoverflow.com/a/32547306/5150303

**Comment by andypotato** [comment]
@daniellivingston setting these entitlements will get your App rejected unless you provide a justification during submission

See details here: https://developer.apple.com/documentation/security/preventing_insecure_network_connections

**Comment by daniellivingston** [comment]
> @daniellivingston setting these entitlements will get your App rejected unless you provide a justification during submission
> 
> 
> 
> See details here: https://developer.apple.com/documentation/security/preventing_insecure_network_connections

Yes, but AFAIK they're the only officially sanctioned way to address this issue. 

At the very least, it's an unblocker.

**Comment by robm92** [comment]
Attempted the above in both info.plist and entitlements with no luck.

**Comment by danielchangsoojones** [comment]
having the same issue here @daniellivingston. Also, when I try and add the NSAllowsArbitraryLoadsForMedia, etc. to my entitlement files, I can't get it to run on my device because it errors with a provisioning profile error:
<img width="1312" alt="Screen Shot 2022-06-30 at 11 57 54 PM" src="https://user-images.githubusercontent.com/16820348/176841387-abaf1726-ad39-493f-82ff-edf514eaca4e.png">
Any solutions?

**Comment by RaffiRincon** [comment]
What helped me was adding this to my Info.plist: 
<img width="680" alt="image" src="https://user-images.githubusercontent.com/10702463/229261707-4d5ab7aa-1d34-4006-a71e-043385699fc8.png">
